### DISCUSSION TITLE
Add support for charting Phidgets HUM1001 data in roast chart for charting stack humidity during roast

### DISCUSSION THREAD

**Discussion by collinarneson** [artisan-roaster-scope/artisan#936] [Ideas]
Hi!

I'm interested in charting the stack humidity during a roast, to measure the water being given off by the coffee as it roasts. This data makes it very obvious when first crack is starting regardless of whether audible cracks are heard or if there is a noticeable effect on ET due to the increase in evaporative cooling at FCs. There are some good examples of humidity data in a roast here: https://www.roastmagazine.com/resources/Articles/2019/2019_Issue1_JanFeb/Roast_JanFeb19_a2_BakedBeans.pdf

I currently have the Phidgets HUM1001 in the exhaust stack of my roaster and have been trying to get some data out of it, but the readings I find on the relevant channel of the Phidgets hub appear to be nonsense. Artisan is able to read the temperature and humidity data streams from this sensor for reporting ambient conditions, so would it be possible to make that data available to add as an extra device for monitoring during a roast?

**Comment by MAKOMO** [comment]
We made some extensive experiments with various moisture sensors (very fast, sensitive and expensive ones that can stand that high temperature and dust) in the exhaust. The results were disappointing. The moisture starts to raise already slowly before FCs and seems not to be indicative to detect FCs. Dave might chime in with more on those results. Note that on a gas roaster the moisture in the exhaust air is also influenced by the amount coming from the burned gas.

The HUM1001 is to measure humidity of the ambient air. It is not fast enough to provide a relevant timeseries data, the high temperatures in the exhaust are outside its specification, and is not sensitive enough for this application.

**Comment by collinarneson** [comment]
Hi Marko, thanks for your response.

With the way my roaster is set up, the stack temperature is within the HUM1001's specifications, and my roaster is electric so I don't need to worry about combustion moisture affecting my readings. I agree that a more sensitive sensor would certainly be desirable for this application, but those are an order of magnitude more expensive and so somewhat impractical to experiment with. I'm not sure if your tests were done with a gas or electric roaster, but on the electric roaster measured in the article I linked, there appears to be a fairly strong signal at FCs in an electric roaster (relative humidity jumps 5% over ~30s at FCs after rising about 1% per minute prior to FCs), so I remain optimistic that this sensor can give useable data in my setup.

<img width="567" alt="Screen Shot 2022-08-11 at 11 49 45" src="https://user-images.githubusercontent.com/83255541/184175500-856b75a5-5a69-4d9d-a765-a23a953418c7.png">

**Reply by MAKOMO** [reply]
As said, this sensor is in several respects not adequate for this application. In case you want to waste time, just patch the Artisan source accordingly.

**Comment by roasterdave** [comment]
Independent of that article I found and used the same sensor.  It is made by Novus Automation and works well.  It is somewhat expensive but can interface with either MODBUS, a 4-20mA current loop or 0-10V so it was easy to connect to Artisan.  Also tried were inexpensive humidity probes readily found on the web that use MODBUS interfaces.  Consulting with an expert at a manufacturer of extremely expensive laboratory grade moisture probes, able to withstand the heat inside the drum, provided much information about the limitations of these devices.  That manufacturer recommended against spending thousands of dollars to use their probe to sense moisture in the drum.  Thus we focused on the exhaust.

Research the chips used to do the actual measurement and you will find there are not very many.  They fall into clear cost/accuracy buckets from which the industrial probes are built.  Look at the operating temperature range and the response time for the humidity sensor.  Response time is only specified at room temperature and a specific air flow so you are on your own to figure what they are at the elevated temperature and air flow in the stack.  Gas changes are sensed because they change the moisture in the exhaust.  While that adds some unwanted perturbations to the signal compared to an electric roaster, these bumps allow measuring the delay in the sensor readings. Having a late indicator is not useful during the roast.  There was a clear difference in performance and construction between the high end probe and the inexpensive ones. 

Also, keep in mind that the particulate matter and any chaff in the exhaust will foul the the sensor.  A good filter built into the probe is essential and it must be replaceable or the probe will have a short life with constantly degrading readings.  

The results trying to identify FC start were, as previously stated, disappointing.  The release of moisture around FC was unmistakable.  However, the ability to use that signal to mark FC start was not reliable.  Correlation to the sound of FC starting or the ETRoR technique was not strong.  The timing and especially the shape of the humidity curve varied too much across a variety of greens to be usable as a marker during the roast.  Post roast the data was interesting but not actionable.  FC could be detected in general but the point in time of its onset could not be reliably identified during or after the roast.  

Maybe you will find a better correlation leading to a reliable FCs marker.  To do so you will need a capable probe designed for this kind of application.

**Reply by collinarneson** [reply]
Thank you for the detailed response, Dave, that's very helpful and gives me a clearer idea of the utility of this data and requirements for the probe in this application.

**Comment by ShakesTC** [comment]
I was thinking about bean humidity during the drying phase. It seems that, within Artisan, this is a part of the roasting process that is left to subjectivity. The roaster determines the end of the phase based on bean color, temperature and experience, circumstantial evidence at best? The accuracy of this end period also influences/sqews the percentages within roast phases, which, according to what I've read, some people rely on to determine their drop times. 

I'm a home roaster using the SmartRoast 800.  I have seen an Artisan chart depicting humidity data within a coffee warehouse storage facility. Do you think I could use that data to more accurately determine the end of the drying phase, and in the end, is this worth pursuing?
Thanks in advance!

**Reply by collinarneson** [reply]
I'm skeptical you could accurately determine the end of the drying phase with that data. It seems like there's a pretty clear humidity spike at 1c (if you can get a humidity sensor that can stand up to the environment and give you high resolution data - fairly big "if"), but not much going on at the end of the drying phase. And more to the point, the key factor here is consistently achieving the dry/ramp/dev ratios that taste good, and it's entirely possible to do that with just (good) temperature data.

**Comment by ShakesTC** [comment]
Thanks for the quick reply. I figured you're right but fooled around with it anyway.  I opened the Roast profile window once per minute, clicked the "Update" button and registered the reading. Sample size was expectedly low and behaved as expected,  higher humidity readings at the beginning of the roast, tapering off at end of drying period, then rising slightly at first crack. I was originally thinking it would be more dramatic and therefore more usable. It was fun to try.
Thanks,
Tom

________________________________
